parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueWriters.java

```diff
@@ -165,9 +178,65 @@ public void writeDouble(int repetitionLevel, double value) {
     }
   }
 
+  private static class FloatWriter extends UnboxedWriter<Float> {
+    private final ColumnDescriptor desc;
+    private long nanCount;
+
+    private FloatWriter(ColumnDescriptor desc, Type.ID id) {
+      super(desc, id);
+      this.desc = desc;
+      this.nanCount = 0;
+    }
+
+    @Override
+    public void write(int repetitionLevel, Float value) {
+      super.write(repetitionLevel, value);
+      if (Float.compare(Float.NaN, value) == 0) {
+        nanCount++;
+      }
+    }
+
+    @Override
+    public Stream<FieldMetrics> metrics() {
+      if (id != null) {
```

Comment:
I think this should always produce the metric. Field IDs are required to write, so we are guaranteed that they are always present (or should fail if one is not). And as long as this is always gathering the metric, we may as well return it.